### ISSUE TITLE
Time-based automatic basetheme switching, again

### ISSUE THREAD

**Issue by The0x539** [ccrama/Slide#1833]
Since the base theme scene has become a bit more complicated since this was last an existent feature in Slide, I imagine it'd be at least a little helpful for me to try to imagine how this would work.

Daytime theme (theme selector)

Nighttime theme (theme selector)

Daytime start (time selector)

Nighttime start(time selector)

Alternately, a timespan selector if that's a thing in some library.

**Comment by The0x539** [comment]
Alternately daytime theme can just be the normally selected base theme.

**Comment by Nxt3** [comment]
I think we're waiting on Android N for this for some reason. The support library already allows this.

**Comment by ccrama** [comment]
I have done this using methods compatible down to ICS, it's a bit reudimentary at the moment but will update soon. You can choose between 4 dark themes to apply at 9pm to 5am, and it will automatically switch to that theme when the time changes.

Only downside to this is the MainActivity has to restart, meaning there is a second of nothing while the theme switches, but IMO this is not a big deal at all (same thing happens when you change theme from the 3 dot menu)

It is also a pro-only feature

**Comment by The0x539** [comment]
Any specifics decided on planned improvements or should somebody make another issue for that?

**Comment by ccrama** [comment]
If you have suggestions, feel free to make another post. Also long time no see!

**Comment by The0x539** [comment]
I'm generally reachable even when I seem to be inactive, FYI. I'll get on
it, hopefully I can finish before some family activities start.
